src/main/scala/mesosphere/mesos/Availability.scala

```diff
@@ -0,0 +1,46 @@
+package mesosphere.mesos
+
+import java.time.Clock
+
+import mesosphere.marathon.RichClock
+import mesosphere.marathon.state.Timestamp
+import org.apache.mesos.Protos.{ DurationInfo, Offer }
+
+import scala.concurrent.duration._
+import scala.language.implicitConversions
+
+object Availability {
+
+  def offerAvailable(offer: Offer, drainingTime: FiniteDuration)(implicit clock: Clock): Boolean = {
+    val now = clock.now()
+    if (offerHasUnavailability(offer)) {
+      val start: Timestamp = offer.getUnavailability.getStart
+
+      if (currentlyInDrainingState(now, start, drainingTime)) {
+        isAgentCurrentlyUnavailable(offer, start, now)
+      } else true
+    } else true
+  }
+
+  private def currentlyInDrainingState(now: Timestamp, start: Timestamp, drainingTime: FiniteDuration) = {
+    now.after(start - drainingTime)
+  }
+
+  private def offerHasUnavailability(offer: Offer) = {
+    offer.hasUnavailability && offer.getUnavailability.hasStart
+  }
+
+  private def isAgentCurrentlyUnavailable(offer: Offer, start: Timestamp, now: Timestamp) = {
```

Comment:
The naming is confusing. Logically it's smth. like `isAgentCurrentlyNOTUnavailable` which sounds bad.